### ISSUE TITLE
FR: nsp, outdated, update and more?

### ISSUE THREAD

**Issue by hemanth** [q-nick/npm-gui#3]
Nice and simple tools like this makes life easy. 

It would be nice to have:

- nsp: show if some packages have security loop holes.
- outdated: highlight the outdated packages.
- update: all bulk update.
- task runner list? (npm script ftw, but it would also be nice to have UI that would allow to manipulate tasks of task runners?)

If yes, I am happy to help!

**Comment by AM-NiceDay** [comment]
@hemanth Update functionality can be really useful for people. Last time I spend quite a lot time to do this manually and this could lead to errors. Do you have any thoughts on this?

**Comment by q-nick** [comment]
@hemanth @AM-NiceDay 

I am glad to hear that someone really likes my tool. Work on the new version just started. (new github stars motivated me)

Now I will focus on features that you requested:
- nsp - i did not know this, but i am totally agree with that feature
- outdated - additional buttons will appear when new version (wanted or latest) is available (then you can just click update now or smth like this)
- support for gulp and grunt could be a overkill for project in situation that you want them to be helpfull (but i will consider them in next steps) 


For now if you want to help - feel free to suggest new features or improvement (i cant predict everything).

**Comment by hemanth** [comment]
@AM-NiceDay [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) helps a lot.

@q-nick True, grunt/glup listing sounds like an overkill now!

**Comment by q-nick** [comment]
@hemanth @AM-NiceDay hey guys, i just updated npm-gui to version 0.2.0, which contains some features that you requested. If you want, please take a look and come back with feedback.

**Comment by hemanth** [comment]
Nice, as of now I notice:

```
Unhandled rejection SyntaxError: Unexpected token / in JSON at position 0
```

```
Unhandled rejection SyntaxError: Unexpected end of JSON input
```

I am sure you would be aware of this, if you could do a todo list, we all can join hands to make a major release 😇

**Comment by q-nick** [comment]
Yeah, few mistakes, for now version 0.2.3 contain a hotfix. 
Some problem can still appear if you have a lot of dependencies in your project. (exec method can reach `exec maxBuffer size`) - fix soon.

**Comment by q-nick** [comment]
Closing. Main features are implemented in application and also `maxBuffer` is fixed (v0.3.0). Support for: Bower, Gulp, Grunt will be added soon.
